### PR TITLE
The First part of Lives system

### LINKED ISSUE #4: Life system
A system to control how many lives each player 
- [X] Ability to give lives if necessary via commands.
- [X] Ability to take lives if necessary via commands.
- [x] Ability to activate and deactivate life system per player.
- [x] All new players start with 3 lives.
- [x] Lives decrease upon death if activated.
- [x] Once lives hit 0, a player becomes a ghost and vice versa.
- [ ] Store lives in a database.
- [ ] Tab should show how many lives each player has.
  - ❓ Potentially it could also be color coded.

### PR DESCRIPTION
**Pull request by MrKacafirekCZ**
This is the first part of lives system. It currently has the most basic way of the system working. Things like storing stuff in a database or showing current lives in tab have not yet been implemented.

### CODE REVIEW

File: src/main/java/io/github/karmasmp/karmaplugin/KarmaPlayer.java [hunk 1]
```diff
@@ -123,4 +141,22 @@ public void setVelocity(Vector vector) {
     public void teleport(Location location) {
         this.player.teleport(location);
     }
+
+    private void updateLives() {
+        if (this.lives > 0 && this.playerLifecycle.getCurrentPhase() instanceof SMPGhostPhase) {
```
Comment (Hoi15A) [inline src/main/java/io/github/karmasmp/karmaplugin/KarmaPlayer.java:146]: If a player starts with 3 lives and we decrement by 1, would this not redundantly tell them that they are now a player even though they already were one?

Also I think updatePlayerPhase or something would make more sense as updateLives does not actually do anything with lives other than read the value.

**Review by Hoi15A — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/karmasmp/karmaplugin/KarmaPlayer.java (hunk 1 above)
Comment (MrKacafirekCZ) [reply src/main/java/io/github/karmasmp/karmaplugin/KarmaPlayer.java:146]: > If a player starts with 3 lives and we decrement by 1, would this not redundantly tell them that they are now a player even though they already were one?

Nope, that's made sure by the `this.playerLifecycle.getCurrentPhase() instanceof SMPGhostPhase/SMPPlayerPhase` condition. By decreasing it to zero or lower while still being in a `SMPPlayerPhase` triggers the correct condition and applies necessary changes such as switching from `SMPPlayerPhase` to a `SMPGhostPhase`.

Comment (Hoi15A) [reply src/main/java/io/github/karmasmp/karmaplugin/KarmaPlayer.java:146]: Ok I see. What about?

> Also I think updatePlayerPhase or something would make more sense as updateLives does not actually do anything with lives other than read the value.

Comment (MrKacafirekCZ) [reply src/main/java/io/github/karmasmp/karmaplugin/KarmaPlayer.java:146]: > Also I think updatePlayerPhase or something would make more sense as updateLives does not actually do anything with lives other than read the value.

I guess that makes more sense.

**Review by Hoi15A — APPROVED** [review]

**Pull request merged**